File: _2024/day-01/problem 2/problem2.py
with open("c:/Users/ppmpr/OneDrive/Documents/GitHub/AOC24/day 1/problem 2/input.txt", 'r') as file:
    data = file.read()

# data = """3   4
# 4   3
# 2   5
# 1   3
# 3   9
# 3   3"""

list_1 = []
list_2 = []

for i in data.split("\n"):
    parts = i.split()
    if len(parts) >= 2:
        list_1.append(int(parts[0]))
        list_2.append(int(parts[1]))

counter_list = [0]*(max(list_2) + 1)

for i in list_2:
    counter_list[i] += 1

final_list = []
for i in range(len(list_1)):
    if list_1[i] < len(counter_list):
        final_list.append(list_1[i]*counter_list[list_1[i]])

print(sum(final_list))

File: _2024/day-01/problem 2/problem2.rs
use std::fs;

fn main(){
    let data: String = fs::read_to_string("C:/Users/ppmpr/OneDrive/Documents/GitHub/AOC24/day-01/problem 2/input.txt").expect("no file found");
    let mut list1: Vec<i32> = Vec::new();
    let mut list2: Vec<i32> = Vec::new();

    for line in data.lines(){
        let temp: Vec<&str> = line.split("   ").collect();
        list1.push(temp[0].parse().unwrap());
        list2.push(temp[1].parse().unwrap());
    }
    
    let mut counter_list: Vec<i32> = Vec::new();
    for _ in 0..*list1.iter().max().unwrap() as usize {
        counter_list.push(0);
    }

    for i in list2{
        counter_list[i as usize]+=1;
    }

    let mut final_list: Vec<i32> = Vec::new();
    for i in 0..list1.len()-1{
        if list1[i] < counter_list.len() as i32 {
            final_list.push(list1[i]*counter_list[list1[i] as usize]);
        }
    }

    println!("{}",final_list.iter().sum::<i32>());
}